crates/ra_syntax/src/grammar/expressions.rs

```diff
@@ -5,6 +5,7 @@ pub(super) use self::atom::{literal, LITERAL_FIRST};
 use super::*;
 
 const EXPR_FIRST: TokenSet = LHS_FIRST;
+const EXPR_FIRST_NO_BLOCK: TokenSet = LHS_FIRST_NO_BLOCK;
```

Comment:
So, this is basically `EXPR_FIRST - {`, right? Let's perhaps just check `EXPR_FIRST.contains(p.current()) && p.current() != L_CURLY`?

Reply:
That's a better way of doing it :man_facepalming:.